browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,236 @@
+      12 - url_->font_list().GetFontSize()));
+}
+
+SplitViewLocationBar::~SplitViewLocationBar() = default;
+
+// static
+SplitViewLocationBar* SplitViewLocationBar::Create(views::View* web_view) {
+  CHECK(web_view->GetWidget());
+  auto* location_bar = new SplitViewLocationBar();
+
+  views::Widget::InitParams params;
+  params.type = views::Widget::InitParams::TYPE_CONTROL;
+  params.parent = web_view->GetWidget()->GetNativeView();
+  params.activatable = views::Widget::InitParams::Activatable::kNo;
+  params.delegate = location_bar;
+  auto* widget = new views::Widget();
+  widget->Init(std::move(params));
+
+  location_bar->view_observation_.Observe(web_view);
+  location_bar->UpdateVisibility();
+  location_bar->UpdateBounds();
+
+  widget->Show();
+
+  return location_bar;
+}
+
+void SplitViewLocationBar::SetWebContents(content::WebContents* web_contents) {
+  if (this->web_contents() == web_contents) {
+    return;
+  }
+
+  Observe(web_contents);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnPaintBackground(gfx::Canvas* canvas) {
+  auto* cp = GetColorProvider();
+  CHECK(cp);
+
+  auto path = GetBorderPath(/*close*/ true);
+  cc::PaintFlags flags;
+  flags.setColor(cp->GetColor(kColorToolbar));
+  flags.setAntiAlias(true);
+  canvas->DrawPath(path, flags);
+}
+
+void SplitViewLocationBar::OnPaintBorder(gfx::Canvas* canvas) {
+  auto* cp = GetColorProvider();
+  CHECK(cp);
+
+  auto path = GetBorderPath(/*close*/ false);
+  cc::PaintFlags flags;
+  flags.setColor(cp->GetColor(kColorBraveSplitViewInactiveWebViewBorder));
+  flags.setAntiAlias(true);
+  flags.setStyle(cc::PaintFlags::kStroke_Style);
+  flags.setStrokeWidth(2);
+  canvas->DrawPath(path, flags);
+}
+
+void SplitViewLocationBar::PrimaryPageChanged(content::Page& page) {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::WebContentsDestroyed() {
+  Observe(nullptr);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnViewVisibilityChanged(views::View* observed_view,
+                                                   views::View* starting_view) {
+  UpdateVisibility();
+}
+
+void SplitViewLocationBar::OnViewBoundsChanged(views::View* observed_view) {
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::OnViewIsDeleting(views::View* observed_view) {
+  view_observation_.Reset();
+}
+
+void SplitViewLocationBar::UpdateVisibility() {
+  auto* view = view_observation_.GetSource();
+  if (view && view->GetVisible()) {
+    GetWidget()->ShowInactive();
+  } else {
+    GetWidget()->Hide();
+  }
+}
+
+void SplitViewLocationBar::UpdateBounds() {
+  auto* view = view_observation_.GetSource();
+  if (!view) {
+    return;
+  }
+
+  gfx::Point point;
+  views::View::ConvertPointToWidget(view, &point);
+
+  auto* widget = GetWidget();
+  CHECK(widget);
+  GetWidget()->SetBounds(gfx::Rect(point, GetPreferredSize()));
+}
+
+void SplitViewLocationBar::UpdateURLAndIcon() {
+  if (auto* contents = web_contents()) {
+    GURL url = contents->GetLastCommittedURL();
```

Comment:
I think other places use `GetVisibleURL` rather than last committed so that the correct URL shows ASAP

Reply:
Nice! But hope security team can double check this, as I was sometimes asked to use committed url.

Reply:
Yup definitely wise :laughing: - I think the reasoning for using `GetVisibleURL` would be that its a visible URL :laughing: (I think its what the Omnibox/CustomTabBar display) but lets defer to the experts :closed_lock_with_key:

Reply:
i think this should always reflect the location of the current page in the webcontents, so GetLastCommittedURL was right